x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/Bucket.java

```diff
@@ -143,7 +137,6 @@ public Bucket(StreamInput in) throws IOException {
         if (in.getVersion().before(Version.V_5_5_0)) {
             in.readGenericValue();
         }
-        partitionScores = in.readList(PartitionScore::new);
```

Comment:
I believe we can get away without doing anything for BWC for the buckets because they are not being transferred between nodes. But I would like @droberts195 to confirm as well.

Reply:
I think we _do_ need to consider BWC for these lists.  If you look at the implementation of `readList()` and `writeList()` they start by reading/writing the list length.  So we need to write an empty list to versions before 6.5, and read a list of something.  We can replace `PartitionScore::new` with a function in `Bucket` that reads the same stuff that `PartitionScore::new` read but just discards it.

Reply:
Thanks @dimitris-athanasiou ! That all makes sense.